stream-rpc/src/connection/client.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+
+use crate::{
+    counters,
+    errors::JsonRpcError,
+    json_rpc::{JsonRpcRequest, JsonRpcResponse, Method, RequestIdentifier},
+    logging,
+    connection::{StreamSender, ConnectionContext, Task},
+};
+use diem_logger::debug;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use storage_interface::DbReader;
+
+#[derive(Debug)]
+pub struct RunningTasks {
+    tasks: Mutex<HashMap<RequestIdentifier, Task>>,
+    client_id: u64,
+}
+
+impl RunningTasks {
+    pub fn new(client_id: &u64) -> Self {
+        Self {
+            tasks: Mutex::new(HashMap::new()),
+            client_id: client_id.clone(),
+        }
+    }
+
+    pub fn stop(&self, id: &RequestIdentifier) -> Option<()> {
+        let mut lock = self.tasks.lock().unwrap();
+        if let Some(task) = lock.get(id) {
+            task.abort();
+            lock.remove(&id);
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    pub fn stop_all(&self) {
+        let tasks = self.tasks.lock().unwrap();
+        debug!(
+            "Aborting {} tasks for Client#{}",
+            tasks.len(),
+            &self.client_id
+        );
+        tasks.iter().for_each(|(_id, task)| task.abort());
+    }
+}
+
+impl Drop for RunningTasks {
+    fn drop(&mut self) {
+        self.stop_all();
+    }
+}
+
+// This will get cloned for every subscription, so lets keep it light :-)
+#[derive(Debug, Clone)]
+pub struct Client {
```

Comment:
When I read `Client` I sort of think about the actual client side making a request to the server. Maybe adding more documentation to explain that this is keeping track of a client connection?